src/components/inline_edit/inline_edit_title.test.tsx

```diff
@@ -10,21 +10,65 @@ import React from 'react';
 import { render } from '../../test/rtl';
 import { requiredProps } from '../../test/required_props';
 
-import { EuiInlineEditTitle } from './inline_edit_title';
+import {
+  EuiInlineEditTitle,
+  EuiInlineEditTitleProps,
+} from './inline_edit_title';
+import { TITLE_SIZES } from '../title/title';
 
 describe('EuiInlineEditTitle', () => {
-  describe('props', () => {
-    test('renders as title', () => {
-      const { container } = render(
-        <EuiInlineEditTitle
-          inputAriaLabel={'textInput'}
-          defaultValue="hello world"
-          heading="h2"
-          {...requiredProps}
-        />
-      );
-
-      expect(container.firstChild).toMatchSnapshot();
+  const inlineEditTitleProps: EuiInlineEditTitleProps = {
+    ...requiredProps,
+    inputAriaLabel: 'Edit title inline',
+    defaultValue: 'Hello World!',
+    heading: 'h1',
+  };
+
+  it('renders in readMode', () => {
+    const { container, getByTestSubject, getByText } = render(
+      <EuiInlineEditTitle {...inlineEditTitleProps} />
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+
+    expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+    expect(getByText('Hello World!')).toBeTruthy();
+  });
+
+  it('renders in editMode', () => {
+    const { container, getByTestSubject, queryByTestSubject } = render(
+      <EuiInlineEditTitle {...inlineEditTitleProps} startWithEditOpen={true} />
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+
+    expect(getByTestSubject('euiInlineEditModeInput')).toBeTruthy();
+    expect(queryByTestSubject('euiInlineReadModeButton')).toBeFalsy();
```

Comment:
You don't need both of these tests - the top level components don't handle readMode or editMode, the internal edit form component does (and already has edit vs read mode thoroughly tested, so there's no need to duplicate it here).

```suggestion
  it('renders', () => {
    const { container, getByTestSubject, getByText } = render(
      <EuiInlineEditTitle {...inlineEditTitleProps} />
    );

    expect(container.firstChild).toMatchSnapshot();
```